backend/src/apiserver/storage/pipeline_store_kubernetes.go

```diff
@@ -0,0 +1,515 @@
+		listOptions = append(listOptions, ctrlclient.InNamespace(common.GetPodNamespace()))
+	}
+
+	err := k.client.List(context.TODO(), &pipelines, listOptions...)
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to find the pipeline")
+	}
+
+	for _, k8sPipeline := range pipelines.Items {
+		if string(k8sPipeline.UID) == pipelineId {
+			return &k8sPipeline, nil
+		}
+	}
+
+	return nil, util.NewResourceNotFoundError("Pipeline", pipelineId)
+}
+
+func (k *PipelineStoreKubernetes) getK8sPipelineVersions(ctx context.Context) (*v2beta1.PipelineVersionList, error) {
+	pipelineVersions := v2beta1.PipelineVersionList{}
+
+	listOptions := []ctrlclient.ListOption{ctrlclient.UnsafeDisableDeepCopy}
+
+	if !common.IsMultiUserMode() && common.GetPodNamespace() != "" {
+		listOptions = append(listOptions, ctrlclient.InNamespace(common.GetPodNamespace()))
+	}
+
+	// Be careful, the deep copy is disabled here to reduce memory allocations
+	err := k.client.List(ctx, &pipelineVersions, listOptions...)
+	if err != nil {
+		return nil, util.NewInternalServerError(
+			err, "Failed to find the pipeline version associated with this pipeline version",
+		)
+	}
+
+	return &pipelineVersions, nil
+}
+
+func (k *PipelineStoreKubernetes) getK8sPipelineVersion(ctx context.Context, pipelineVersionId string) (*v2beta1.PipelineVersion, error) {
+	pipelineVersions, err := k.getK8sPipelineVersions(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, k8sPipelineVersion := range pipelineVersions.Items {
+		if string(k8sPipelineVersion.UID) == pipelineVersionId {
+			return &k8sPipelineVersion, nil
+		}
+	}
+
+	return nil, util.NewResourceNotFoundError("PipelineVersion", pipelineVersionId)
+}
+
+func (k *PipelineStoreKubernetes) createPipelineVersionWithPipeline(ctx context.Context, pipeline *model.Pipeline, pipelineVersion *model.PipelineVersion) (*model.PipelineVersion, error) {
+	if pipelineVersion.PipelineSpecURI != "" {
+		return nil, util.NewBadRequestError(ErrUnsupportedField, "The pipeline spec URI field is not supported")
+	}
+
+	k8sPipelineVersion, err := v2beta1.FromPipelineVersionModel(*pipeline, *pipelineVersion)
+	if err != nil {
+		return nil, util.NewBadRequestError(err, "Invalid pipeline spec")
+	}
+
+	pipelineVersionStatus := k8sPipelineVersion.Status.DeepCopy()
+
+	glog.Infof(
+		"Creating the pipeline version %s/%s in Kubernetes", k8sPipelineVersion.Namespace, k8sPipelineVersion.Name,
+	)
+	err = k.client.Create(ctx, k8sPipelineVersion)
+	if k8serrors.IsAlreadyExists(err) {
+		return nil, util.NewAlreadyExistError(
+			"Failed to create a new pipeline version. The name %v already exists. Please specify a new name",
+			pipelineVersion.Name,
+		)
+	} else if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to create the pipeline version")
+	}
+
+	k8sPipelineVersion.Status = *pipelineVersionStatus
+
+	err = k.client.Status().Update(ctx, k8sPipelineVersion)
```

Comment:
Did you determine if this could be removed since it seems the status will always be empty here?